### ISSUE TITLE
cloudforms pod running  , app start Failure

### ISSUE THREAD

**Issue by lixwlixw** [openshift/openshift-ansible#5515]
[root@master1 cloudforms]# oc get pod
NAME                 READY     STATUS    RESTARTS   AGE
cloudforms-0         0/1       Running   0          1m
memcached-1-1rmqg    1/1       Running   0          22m
postgresql-3-1mjvw   1/1       Running   0          5m
[root@master1 cloudforms]# oc logs -f cloudforms-0 

cloudforms app not start , no log .

[root@master1 cloudforms]# oc describe pod cloudforms-0 
Name:			cloudforms-0
Namespace:		cloudforms
Security Policy:	privileged
Node:			node5/10.161.34.14
Start Time:		Mon, 25 Sep 2017 14:09:41 +0800
Labels:			name=cloudforms
Status:			Running
IP:			172.31.2.52
Controllers:		StatefulSet/cloudforms
Containers:
  cloudforms:
    Container ID:	docker://b96e2139742e641b82deacc452d0e16044c2195dc0bca37730f753090cd15940
    Image:		registry.access.redhat.com/cloudforms45/cfme-openshift-app:latest
    Image ID:		docker-pullable://registry.access.redhat.com/cloudforms45/cfme-openshift-app@sha256:bc6baac5aeba5affe0bada1bfbe330cd2d58da82767d66b3fa9ab12471a1b0f5
    Ports:		80/TCP, 443/TCP
    Limits:
      memory:	16Gi
    Requests:
      cpu:		1
      memory:		6Gi
    State:		Running
      Started:		Mon, 25 Sep 2017 14:10:00 +0800
    Ready:		False
    Restart Count:	0
    Liveness:		tcp-socket :443 delay=480s timeout=3s period=10s #success=1 #failure=3
    Readiness:		http-get https://:443/ delay=200s timeout=3s period=10s #success=1 #failure=3
    Volume Mounts:
      /persistent from cloudforms-server (rw)
      /persistent-region from cloudforms-region (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-x7dvm (ro)
    Environment Variables:
      APPLICATION_INIT_DELAY:		15
      DATABASE_SERVICE_NAME:		postgresql
      DATABASE_REGION:			0
      MEMCACHED_SERVICE_NAME:		memcached
      POSTGRESQL_USER:			root
      POSTGRESQL_PASSWORD:		smartvm
      POSTGRESQL_DATABASE:		vmdb_production
      POSTGRESQL_MAX_CONNECTIONS:	100
      POSTGRESQL_SHARED_BUFFERS:	256MB
Conditions:
  Type		Status
  Initialized 	True 
  Ready 	False 
  PodScheduled 	True 
Volumes:
  cloudforms-server:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	cloudforms-server-cloudforms-0
    ReadOnly:	false
  cloudforms-region:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	cloudforms-region
    ReadOnly:	false
  default-token-x7dvm:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	default-token-x7dvm
QoS Class:	Burstable
Tolerations:	<none>
Events:
  FirstSeen	LastSeen	Count	From			SubObjectPath			Type		Reason		Message
  ---------	--------	-----	----			-------------			--------	------		-------
  4m		4m		1	{default-scheduler }					Normal		Scheduled	Successfully assigned cloudforms-0 to node5
  4m		4m		1	{kubelet node5}		spec.containers{cloudforms}	Normal		Pulling		pulling image "registry.access.redhat.com/cloudforms45/cfme-openshift-app:latest"
  4m		4m		1	{kubelet node5}		spec.containers{cloudforms}	Normal		Pulled		Successfully pulled image "registry.access.redhat.com/cloudforms45/cfme-openshift-app:latest"
  4m		4m		1	{kubelet node5}		spec.containers{cloudforms}	Normal		Created		Created container with docker id b96e2139742e; Security:[seccomp=unconfined]
  4m		4m		1	{kubelet node5}		spec.containers{cloudforms}	Normal		Started		Started container with docker id b96e2139742e
  55s		5s		6	{kubelet node5}		spec.containers{cloudforms}	Warning		Unhealthy	Readiness probe failed: Get https://172.31.2.52:443/: dial tcp 172.31.2.52:443: getsockopt: connection refused

Who can help me ?

**Comment by aizuddin85** [comment]
@lixwlixw `oc get dc -n cloudforms` and `oc describe dc/<dc> -n cloudforms' also 'oc status -v -n cloudforms' please.

**Comment by lixwlixw** [comment]
<pre><code>[root@master1 ~]# oc get dc -n cloudforms
NAME         REVISION   DESIRED   CURRENT   TRIGGERED BY
memcached    1          1         1         config,image(cfme-openshift-memcached:latest)
postgresql   5          1         1         config,image(cfme-openshift-postgresql:latest)
</code></pre>

<pre><code>[root@master1 ~]# oc describe dc memcached
Name:		memcached
Namespace:	cloudforms
Created:	24 hours ago
Labels:		app=cloudforms
		template=cloudforms
Description:	Defines how to deploy memcached
Annotations:	openshift.io/generated-by=OpenShiftNewApp
Latest Version:	1
Selector:	name=memcached
Replicas:	1
Triggers:	Image(cfme-openshift-memcached@latest, auto=true), Config
Strategy:	Recreate
Template:
  Labels:	app=cloudforms
		name=memcached
  Annotations:	openshift.io/generated-by=OpenShiftNewApp
  Containers:
   memcached:
    Image:	registry.access.redhat.com/cloudforms45/cfme-openshift-memcached@sha256:8f4b1280641deed59e50496c30f3a3fd8c422f4c2912f197db006c664f53fcc0
    Port:	11211/TCP
    Limits:
      memory:	256Mi
    Requests:
      cpu:		200m
      memory:		64Mi
    Liveness:		tcp-socket :11211 delay=30s timeout=1s period=10s #success=1 #failure=3
    Readiness:		tcp-socket :11211 delay=5s timeout=1s period=10s #success=1 #failure=3
    Volume Mounts:	<none>
    Environment Variables:
      MEMCACHED_MAX_MEMORY:		64
      MEMCACHED_MAX_CONNECTIONS:	1024
      MEMCACHED_SLAB_PAGE_SIZE:		1m
  No volumes.

Deployment #1 (latest):
	Name:		memcached-1
	Created:	24 hours ago
	Status:		Complete
	Replicas:	1 current / 1 desired
	Selector:	deployment=memcached-1,deploymentconfig=memcached,name=memcached
	Labels:		app=cloudforms,openshift.io/deployment-config.name=memcached,template=cloudforms
	Pods Status:	1 Running / 0 Waiting / 0 Succeeded / 0 Failed

No events.
</code></pre>

<pre><code>[root@master1 ~]# oc describe dc postgresql 
Name:		postgresql
Namespace:	cloudforms
Created:	24 hours ago
Labels:		app=cloudforms
		template=cloudforms
Description:	Defines how to deploy the database
Annotations:	openshift.io/generated-by=OpenShiftNewApp
Latest Version:	5
Selector:	name=postgresql
Replicas:	1
Triggers:	Image(cfme-openshift-postgresql@latest, auto=true), Config
Strategy:	Recreate
Template:
  Labels:	app=cloudforms
		name=postgresql
  Annotations:	openshift.io/generated-by=OpenShiftNewApp
  Containers:
   postgresql:
    Image:	registry.access.redhat.com/cloudforms45/cfme-openshift-postgresql@sha256:15e7703d79ecf0dfba662a84e300567240769d5802011f75a95fad2b25b24b89
    Port:	5432/TCP
    Limits:
      memory:	8Gi
    Requests:
      cpu:	500m
      memory:	1Gi
    Liveness:	tcp-socket :5432 delay=60s timeout=1s period=10s #success=1 #failure=3
    Readiness:	exec [/bin/sh -i -c psql -h 127.0.0.1 -U ${POSTGRESQL_USER} -q -d ${POSTGRESQL_DATABASE} -c 'SELECT 1'] delay=15s timeout=1s period=10s #success=1 #failure=3
    Volume Mounts:
      /var/lib/pgsql/data from cfme-pgdb-volume (rw)
    Environment Variables:
      POSTGRESQL_USER:			root
      POSTGRESQL_PASSWORD:		smartvm
      POSTGRESQL_DATABASE:		vmdb_production
      POSTGRESQL_MAX_CONNECTIONS:	100
      POSTGRESQL_SHARED_BUFFERS:	256MB
  Volumes:
   cfme-pgdb-volume:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	cloudforms-postgresql
    ReadOnly:	false

Deployment #5 (latest):
	Name:		postgresql-5
	Created:	2 hours ago
	Status:		Complete
	Replicas:	1 current / 1 desired
	Selector:	deployment=postgresql-5,deploymentconfig=postgresql,name=postgresql
	Labels:		app=cloudforms,openshift.io/deployment-config.name=postgresql,template=cloudforms
	Pods Status:	1 Running / 0 Waiting / 0 Succeeded / 0 Failed
Deployment #4:
	Created:	2 hours ago
	Status:		Complete
	Replicas:	0 current / 0 desired
Deployment #3:
	Created:	23 hours ago
	Status:		Complete
	Replicas:	0 current / 0 desired

Events:
  FirstSeen	LastSeen	Count	From				SubObjectPath	Type		Reason			Message
  ---------	--------	-----	----				-------------	--------	------			-------
  1h		1h		1	{deploymentconfig-controller }			Normal		DeploymentCreated	Created new replication controller "postgresql-5" for version 5
</code></pre>

<pre><code>[root@master1 ~]# oc status -v

  statefulset/cloudforms manages registry.access.redhat.com/cloudforms45/cfme-openshift-app:latest, created 24 hours ago - 1 pod

svc/memcached - 172.30.38.158:11211
  dc/memcached deploys istag/cfme-openshift-memcached:latest 
    deployment #1 deployed 24 hours ago - 1 pod

svc/postgresql - 172.30.228.18:5432
  dc/postgresql deploys istag/cfme-openshift-postgresql:latest 
    deployment #5 deployed 2 hours ago - 1 pod
    deployment #4 deployed 2 hours ago
    deployment #3 deployed 23 hours ago

Warnings:
  * pod/cloudforms-0 has restarted within the last 10 minutes

View details with 'oc describe <resource>/<name>' or list everything with 'oc get all'.
</code></pre>


<pre><code>[root@master1 ~]# oc describe statefulset cloudforms 
Name:			cloudforms
Namespace:		cloudforms
Image(s):		registry.access.redhat.com/cloudforms45/cfme-openshift-app:latest
Selector:		name=cloudforms
Labels:			app=cloudforms,template=cloudforms
Replicas:		1 current / 1 desired
Annotations:		description=Defines how to deploy the CloudForms appliance,openshift.io/generated-by=OpenShiftNewApp
CreationTimestamp:	Mon, 25 Sep 2017 13:48:53 +0800
Pods Status:		1 Running / 0 Waiting / 0 Succeeded / 0 Failed
Volumes:
  cloudforms-region:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	cloudforms-region
    ReadOnly:	false
No events.
</code></pre>

**Comment by aizuddin85** [comment]
@lixwlixw somehow the cloudfrom dc is not exists for your deployment. This is on my test environment:

```
[root@master /]# oc get dc -n cfme
NAME         REVISION   DESIRED   CURRENT   TRIGGERED BY
cloudforms   1          1         0         config,image(cfme-openshift-app:latest)
memcached    1          1         1         config,image(cfme-openshift-memcached:latest)
postgresql   2          1         1         config,image(cfme-openshift-postgresql:latest)
```

Note: Desired is 0 since its deploying.

**Comment by lixwlixw** [comment]
What I'm referring to is :   <https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_examples/files/examples/v1.5/cfme-templates/>

Where did your deployment file come from?

**Comment by aizuddin85** [comment]
@lixwlixw https://access.redhat.com/documentation/en-us/red_hat_cloudforms/4.2/html/installing_red_hat_cloudforms_on_openshift_container_platform/

https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_examples/files/examples/v1.4/cfme-templates

**Comment by lixwlixw** [comment]
I sent you an email（aizuddin.zali@gmail.com）。

**Comment by sdodson** [comment]
/assign tbielawa
We don't need to fix this in current code but we do need to make sure this is addressed in your 4.6 work
